manifests/init.pp

```diff
@@ -1,16 +1,108 @@
 # A class to install and manage Yum configuration.
 #
-# @param clean_old_kernels Whether or not to purge old kernel version beyond the `keeponly_limit`.
-# @param keep_kernel_devel Whether or not to keep kernel devel packages on old kernel purge.
-# @param config_options A Hash where keys are the names of `Yum::Config` resources and the values
+# @param clean_old_kernels [Boolean]
+#   Whether or not to purge old kernel version beyond the `keeponly_limit`.
+#
+# @param keep_kernel_devel [Boolean]
+#   Whether or not to keep kernel devel packages on old kernel purge.
+#
+# @param config_options [Hash]
+#   A Hash where keys are the names of `Yum::Config` resources and the values
 #   are either the direct `ensure` value, or a Hash of the resource's attributes.
-#   @note Boolean values will be converted to either a `1` or `0`; use a quoted string to get a
-#     literal `true` or `false`.
+#
+#   @note Boolean parameter values will be converted to either a `1` or `0`; use a quoted string to
+#     get a literal `true` or `false`.
+#
+# @param repos [Hash]
+#   A hash where keys are the names of `Yumrepo` resources and each value represents its respective
+#   Yumrepo's resource parameters.  This is used in conjunction with the `managed_repos` parameter
+#   to create `Yumrepo` resources en masse.  Some default data is provided for this using module
+#   data.  It is configured to deep merge with a `knockout_prefix` of `--` by default, so individual
+#   parameters may be overriden or removed via global or environment Hiera data.
+#
+#   @note Boolean parameter values will be converted to either a `1` or `0`; use a quoted string to
+#     get a literal `true` or `false`.
+#
+# @param managed_repos [Array]
+#   An array of first-level keys from the `repos` hash to include in the catalog.  The module uses
+#   this list to select `Yumrepo` resources from the `repos` hash for instantiation.  Defaults are
+#   set in the module's Hiera data.
+#
+#   @note This only indicates the *managed* state of the repos, the `ensure` state must be managed
+#     in the `repos` data.
+#
+# @param manage_os_default_repos [Boolean]
+#   Whether or not to add an operating system's default repos to the `managed_repos` array.
+#
+#   @note Currently only valid on CentOS.
```

Comment:
This parameter is valid on other OSs, but `os_default_repos` will provide no defaults for any OS other than CentOS 6/7. Perhaps say something like "currently only provides defaults for CentOS" on the `os_default_repos` parameter instead?